__tests__/roundtrip_test.ml

```diff
@@ -13,6 +11,40 @@ let run_test ~name ~write ~read ~data =
     |> toEqual data'
   )
 
+let run_read_test ~name ~read ~data ~expected =
+  let open Expect in
+  let decode = Atdgen_codec_runtime.Decode.decode read in
+  let data' = decode data in
+  test name (fun () ->
+    expect data'
+    |> toEqual expected
+  )
+
+
+(** Test that decoding data that wouldn't normally be encoded still equals the correct value *)
+let () =
+    describe "read tests" (fun () ->
+      run_read_test
+        ~name:"nullable field set to null"
+        ~read:Test_bs.read_optional_field
+        ~expected:{with_default = 9; no_default = None; no_default_nullable = None}
+        ~data:begin
+          let input = Js.Dict.empty () in
+          let () = Js.Dict.set input "no_default_nullable" Js.Json.null in
+          Js.Json.object_ input
+        end;
+      run_read_test
+        ~name:"optional field set to null"
+        ~read:Test_bs.read_optional_field
+        ~expected:{with_default = 9; no_default = None; no_default_nullable = None}
+        ~data:begin
+          let input = Js.Dict.empty () in
+          let () = Js.Dict.set input "no_default" Js.Json.null in
+          Js.Json.object_ input
+        end;
```

Comment:
@TheSpyder @Khady do you think it'd be worth to add tests for `keep_nulls` in the runtime?

Reply:
The bucklescript runtime nor atdgen's bucklescript generator doesn't really seem to support `keep_nulls` - but perhaps they should - this change just makes the runtime assume the default value.